Even more smart pointers for local variables in WebCore

annevk opened the pull request:
#### 6b87ca2de3136db2180c50cbdd3ccaa43a681cac
<pre>
Even more smart pointers for local variables in WebCore
<a href="https://bugs.webkit.org/show_bug.cgi?id=294728">https://bugs.webkit.org/show_bug.cgi?id=294728</a>

Reviewed by Charlie Wolfe.

Apply <a href="https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines">https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines</a>

Canonical link: <a href="https://commits.webkit.org/296475@main">https://commits.webkit.org/296475@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/20103a31648c56dc67c9af88a64e761fe99b71aa

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/108633 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/28294 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/18718 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/113841 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/59018 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/110596 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/28983 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/36849 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/82516 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/111581 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/23011 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/97855 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/62953 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/22428 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/15992 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/58552 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/92381 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/16043 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/116962 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/35687 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/26317 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/91538 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/36060 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/94124 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/91342 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/36245 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/14007 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/31510 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/17541 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/35588 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/41121 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/35298 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/38644 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/36975 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

charliewolfe commented on Source/WebCore/bindings/js/WebCoreJSClientData.cpp:
```diff
@@ -175,7 +175,7 @@ void JSVMClientData::getAllWorlds(Vector<Ref<DOMWrapperWorld>>& worlds)
 
     // Add other normal worlds.
-    for (auto* world : m_worldSet) {
+    for (RefPtr world : m_worldSet) {
```
use releaseNonNull() when appending to `worlds` below

charliewolfe commented on Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
```diff
@@ -139,18 +139,18 @@ static FragmentAndResources createFragmentInternal(LocalFrame&, NSAttributedStri
-    auto fragment = DocumentFragment::create(document);
-    auto dummyBodyToForceInBodyInsertionMode = HTMLBodyElement::create(document);
+    auto fragment = DocumentFragment::create(document.get());
+    auto dummyBodyToForceInBodyInsertionMode = HTMLBodyElement::create(document.get());
```
```suggestion
    Ref fragment = DocumentFragment::create(document.get());
    Ref dummyBodyToForceInBodyInsertionMode = HTMLBodyElement::create(document.get());
```

charliewolfe commented on Source/WebCore/editing/cocoa/WebContentReaderCocoa.mm:
```diff
@@ -495,8 +498,8 @@ static void simplifyFragmentForSingleTextAttachment(NSAttributedString *string,
         Ref data = subresource->data();
-        auto blob = Blob::create(&document, data->copyData(), subresource->mimeType());
-        String blobURL = DOMURL::createObjectURL(document, blob);
+        auto blob = Blob::create(document.get(), data->copyData(), subresource->mimeType());
```
```suggestion
        Ref blob = Blob::create(document.get(), data->copyData(), subresource->mimeType());
```

charliewolfe commented on Source/WebCore/inspector/WebInjectedScriptManager.cpp:
```diff
@@ -77,7 +77,7 @@ void WebInjectedScriptManager::discardInjectedScriptsFor(LocalDOMWindow& window)
         return;
 
-    auto* document = window.document();
+    RefPtr document = window.document();
```
Seems like a false positive from the static analyzer — this usage is safe. No big deal to use RefPtr though.

cc @rniwa

charliewolfe commented on Source/WebCore/inspector/agents/WebConsoleAgent.cpp:
```diff
@@ -64,9 +64,9 @@ WebConsoleAgent::WebConsoleAgent(WebAgentContext& context)
 void WebConsoleAgent::frameWindowDiscarded(LocalDOMWindow& window)
 {
-    if (auto* document = window.document()) {
+    if (RefPtr document = window.document()) {
```
Same thing here, usage is safe (again, adding RefPtr seems ok anyways)

annevk committed:
Even more smart pointers for local variables in WebCore
https://bugs.webkit.org/show_bug.cgi?id=294728

Reviewed by Charlie Wolfe.

Apply https://github.com/WebKit/WebKit/wiki/Safer-CPP-Guidelines

Canonical link: https://commits.webkit.org/296475@main
